src/meta/src/barrier/recovery.rs

```diff
@@ -387,6 +373,12 @@ impl GlobalBarrierManagerContext {
                         warn!(err = ?err, "reset compute nodes failed");
                     })?;
 
+                    let to_remove_actors = scheduled_barriers.pre_apply_drop_scheduled().await;
+                    info.post_apply(Some(CommandActorChanges {
```

Comment:
I would suggest here we get the globally synced actor info again later via `self.resolve_actor_info().await` when we create a new `BarrierManagerState` instead of incrementally applying the change here.